### DISCUSSION TITLE
How to return the schema in json for frontends to use with graphql codegen?

### DISCUSSION THREAD

**Discussion by khuongtp** [99designs/gqlgen#2951] [Q&A] [answered]
Hi folks! Can somebody please tell me how to return the schema in json so that I can config the graphql codgen from frontend to use the url from go server. I'm pretty new to go. Thanks in advanced!

**Comment by UnAfraid** [comment, marked as answer]
You can use integration testing package as an example of how to achieve this.
https://github.com/99designs/gqlgen/tree/master/integration

Codegen configuration points to the graphql endpoint where it would perform introspection
https://github.com/99designs/gqlgen/blob/master/integration/codegen.ts

Here you can see the command that generates the code `npm run gen`
https://github.com/99designs/gqlgen/blob/master/integration/package.json#L8

The graphql code generator is scanning src/**/*.graphql for your queries, mutations and subscriptions

**Comment by WillSkelton** [comment]
@khuongtp Did you ever get this to work? I'm trying to do the same thing and I'm having trouble.

**Reply by khuongtp** [reply]
Yes I actually did, here is my config from FE, very simple, what's important is the schema field
```ts
const config: CodegenConfig = {
  overwrite: true,
  schema: "http://localhost:8080/query",
  ignoreNoDocuments: true,
  documents: ["src/graphql/**/*.graphql", "src/graphql/documents/**/*.ts"],
  generates: {
    "src/graphql/types.ts": {
      plugins: ["typescript", "typescript-operations"],
    },
  },
};
```
and here is what it looks like from BE 
```go
srv.Use(extension.Introspection{})
router.Handle("/", playground.Handler("GraphQL playground", "/query"))
router.Handle("/query", srv)
```
It just works out of the box if you follow the example 😃 , I believe the introspection extension is needed if you don't use the handler.NewDefaultServer as in the example

**Reply by WillSkelton** [reply]
Oh heck yeah dude. That works. Thank you so much!
